src/pghstore/_native.py

```diff
@@ -227,12 +227,14 @@ def parse(string, encoding='utf-8'):
 
     .. sourcecode:: pycon
 
-       >>> list(parse('a=>1, b => 2, c => null, d => "NULL"'))
-       [(u'a', u'1'), (u'b', u'2'), (u'c', None), (u'd', u'NULL')]
-       >>> list(parse(r'"a=>1"=>"\"b\"=>2",'))
-       [(u'a=>1', u'"b"=>2')]
+       >>> list(parse('a=>1, b => 2, c => null, d => "NULL"')) == [(u'a', u'1'), (u'b', u'2'), (u'c', None), (u'd', u'NULL')]
+       True
+       >>> list(parse(r'"a=>1"=>"\"b\"=>2",')) == [(u'a=>1', u'"b"=>2')]
+       True
 
     """
+    if isinstance(string, six.binary_type):
+        string = string.decode(encoding)
```

Comment:
possibly handling unicode except at the edges is better overall. Less fiddling with encoding and conditionals.